### PR TITLE
Added the EMNIST dataset

### PR DESCRIPTION
**Pull request by JoshuaBradbury**
This PR adds EMNIST to the datasets. EMNIST is an extension of the MNIST dataset but it also adds Characters, and more numbers. It can be found in the paper here: https://arxiv.org/abs/1702.05373v1

### CODE REVIEW

**Comment by av8ramit** [comment]
Jenkins, test this please.

**Comment by JoshuaBradbury** [comment]
The problem in the initial test was down to it taking too long, but this is mainly caused by where the data is pulled from, as well as the size of the data. This problem wasn't really a concern for the MNIST set as it only had at most 70,000 images and labels, where as every set in the EMNIST is bigger, and can be up to 800,000 images and labels.

**Comment by JoshuaBradbury** [comment]
@tensorflow-jenkins test this please.

File: tensorflow/contrib/learn/python/learn/datasets/emnist.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
+# Copyright 2017 Joshua Bradbury. All Rights Reserved.
```
Comment (drpngx) [inline tensorflow/contrib/learn/python/learn/datasets/emnist.py]: We usually have "The TensorFlow Authors" here. Any to change it to that?

**Comment by nealwu** [comment]
@vincentvanhoucke is this a dataset we are interested in adding? I hadn't heard of it before this.

**Comment by vincentvanhoucke** [comment]
@nealwu I don't know what's our policy in regards to accepting new datasets into TF.
This one only has 2 citations thus far: https://scholar.google.com/scholar?oi=bibs&hl=en&cites=8838276906904533917&as_sdt=5

**Comment by nealwu** [comment]
Doesn't seem noteworthy enough to include then. Going to close this. If others think it is worth including, feel free to reopen.

**Pull request closed without merging**